### PR TITLE
Pooling interface rework

### PR DESCRIPTION
**Pull request by W3stside**
## Refactors Pooling Interface

1. 2 steps (step 2 and 3 were so similar)
2. useForm and schema validation for inputs
3. moved files out (sorry about that - lots of noise)

I respectfully request your forgiveness for the size of the PR senseis *bows-deeply.jpeg*

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr683--dexreact.review.gnosisdev.com

File: src/components/PoolingWidget/DefineSpread.tsx [hunk 1]
```diff
@@ -1,52 +1,48 @@
...
+        <small>percentage you want to sell above $1, and buy below $1 between all selected tokens</small>
+      </p>
+      <InputWithTooltip
+        name="spread"
+        type="number"
+        step={0.1}
+        disabled={isSubmitting}
+        ref={register}
+        onKeyPress={onKeyPress}
+        tooltip={(errorMessage && formatErrorMessage(errorMessage)) || 'Value between 0 and 100, not inclusive'}
```
Comment (Velenir) [inline src/components/PoolingWidget/DefineSpread.tsx]: I'm not sure it's a good idea to use tooltips to report form errors. Not without styling tooltips in red colors or something, at least

File: src/components/PoolingWidget/StepDecriptors.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        }
+      default:
+        return { title: 'An error occurred, please try again' }
+    }
+  }, [step])
+
+  return (
+    <div>
+      <ProgressStepText as="h2">{title}</ProgressStepText>
+      {subtext && <div className="liqContent" dangerouslySetInnerHTML={{ __html: subtext }} />}
```
Comment (Velenir) [inline src/components/PoolingWidget/StepDecriptors.tsx]: `dangerouslySetInnerHTML` Please don't!:fearful:
Why is it even here?!:cold_sweat: Why was this added?!:worried: Why didn't I notice?! :confounded:

File: src/components/PoolingWidget/index.tsx [hunk 3]
```diff
@@ -192,6 +112,23 @@ const PoolingInterface: React.FC = () => {
   // Avoid displaying an empty list of tokens when the wallet is not connected
   const fallBackNetworkId = networkId ? networkId : Network.Mainnet // fallback to mainnet
 
+  const methods = useForm({
+    defaultValues: {
+      // @ts-ignore
```
Comment (Velenir) [inline src/components/PoolingWidget/index.tsx]: eh? You probably need to type `useForm<>`

File: src/components/PoolingWidget/StepDecriptors.tsx (hunk 2 above)
Comment (W3stside) [reply src/components/PoolingWidget/StepDecriptors.tsx]: didnt add this, not sure who did, maybe @biocom did for some testing?

File: src/components/PoolingWidget/index.tsx (hunk 3 above)
Comment (W3stside) [reply src/components/PoolingWidget/index.tsx]: oops

File: src/components/PoolingWidget/DefineSpread.tsx (hunk 1 above)
Comment (W3stside) [reply src/components/PoolingWidget/DefineSpread.tsx]: done, good advice

File: src/components/PoolingWidget/DefineSpread.tsx [hunk 4]
```diff
@@ -1,52 +1,48 @@
...
-const DefineSpread: React.FC<DefineSpreadProps> = ({ selectedTokensMap, spread, setSpread }) => {
-  const handleSpreadChange = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>): void => {
-    if (+value < 0 || +value >= 100) return
+const DefineSpread: React.FC<DefineSpreadProps> = ({ isSubmitting }) => {
+  const { errors, register } = useFormContext()
 
-    setSpread(+value)
-  }
+  const onKeyPress = useCallback((event: React.KeyboardEvent<HTMLInputElement>): void => {
+    if (event.key === 'Enter') {
```
Comment (alfetopito) [inline src/components/PoolingWidget/DefineSpread.tsx]: Can I haz enter to submit form?

Comment (alfetopito) [reply src/components/PoolingWidget/DefineSpread.tsx]: I still can't `enter` on spread :(

Comment (W3stside) [reply src/components/PoolingWidget/DefineSpread.tsx]: works for me on FF, whatre u using

Comment (W3stside) [reply src/components/PoolingWidget/DefineSpread.tsx]: the PR URL hasnt updated due to the develop mismatch, ill merge and then it should work

Comment (alfetopito) [reply src/components/PoolingWidget/DefineSpread.tsx]: Working now \o/ yay!

File: src/utils/validation.ts [hunk 5]
```diff
@@ -16,31 +17,33 @@ export function validatePositive(value: string, constraint = 0): true | string {
...
     type === 'number'
       ? Object.keys(data).reduce((acc, key) => {
           const oldValue = data[key]
           const castedValue = Number(oldValue)
-          const castedObj = { ...acc, [key]: castedValue }
+          const castedObj: D = { ...acc, [key]: castedValue }
 
           return castedObj
-        }, {})
+        }, {} as D)
```
Comment (Velenir) [inline src/utils/validation.ts]: I'm not sure of it but try `Object.keys(data).reduce<D>` instead of `as D`
If doesn't help, then whatever

File: src/utils/validation.ts [hunk 6]
```diff
@@ -16,31 +17,33 @@ export function validatePositive(value: string, constraint = 0): true | string {
...
-    values: error ? {} : values,
+    values: error || !value ? {} : data,
     errors: error
       ? error.details.reduce((previous, currentError) => {
           return {
             ...previous,
             [currentError.path[0]]: currentError,
           }
-        }, {})
+        }, {} as ValidationError)
```
Comment (Velenir) [inline src/utils/validation.ts]: Try `.reduce<Type>` trick

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr683--dexreact.review.gnosisdev.com
